### PR TITLE
Support multi trie in genesis generation

### PR DESCRIPTION
**Pull request by sorpaas**
This changes `BuildStorage` from returning `StorageMap` to `(StorageMap, ChilrenStorageMap)`, where

* `StorageMap = HashMap<Vec<u8>, Vec<u8>>`
* `ChildrenStorageMap = HashMap<Vec<u8>, StorageMap>`

Genesis blocks are generated with children storage, if provided. In `decl_storage`, the build storage call is changed to accept another parameter `ChildrenStorageMap` to insert children mappings.

### CODE REVIEW

**Comment by gavofyork** [comment]
seems reasonable. would get a couple of more reviews before merge. @arkpar ?

File: core/client/db/src/lib.rs [hunk 1]
```diff
@@ -104,10 +104,10 @@ pub fn new_client<E, S, Block>(
 	block_execution_strategy: ExecutionStrategy,
 	api_execution_strategy: ExecutionStrategy,
 ) -> Result<client::Client<Backend<Block>, client::LocalCallExecutor<Backend<Block>, E>, Block>, client::error::Error>
-	where
-		Block: BlockT,
-		E: CodeExecutor<Blake2Hasher> + RuntimeInfo,
-		S: BuildStorage,
+where
+	Block: BlockT<Hash=H256>,
```
Comment (arkpar) [inline core/client/db/src/lib.rs]: Where is this requirement coming from? It is unfortunate to see it propagating all the way up.

Comment (sorpaas) [reply core/client/db/src/lib.rs]: It's from the previous-implicit requirement that `H::Out == Block::Hash`. As we have been using `Blake2Hasher` specialization, that means in many places we're practically already limited to `H256`. So I don't think we've lost generalization over this.

**Comment by sorpaas** [comment]
CI seems to have broken somehow. Let me see whether close and reopen would help.

**Pull request merged**